pkg/spireentry/reconciler.go

```diff
@@ -630,3 +640,31 @@ func idsFromEntries(entries []spireapi.Entry) []string {
 	}
 	return ids
 }
+
+// filterJoinTokenEntries filters out entries that correspond to join tokens.
+func filterJoinTokenEntries(entries []spireapi.Entry) []spireapi.Entry {
+	var filteredEntries []spireapi.Entry
+	for _, entry := range entries {
+		if strings.Contains(entry.ParentID.String(), joinTokenSpiffeSubstr) {
+			continue
+		}
+		filteredEntries = append(filteredEntries, entry)
+	}
+	return filteredEntries
+}
+
+// isJoinTokenEntry returns true if the entry corresponds to a join token.
+// For an entry to correspond to a join token, both the following conditions must be true:
+// 1. The parent ID of the entry must contain the substring "/spire/agent/join_token/".
+// 2. The entry must contain a selector of type "spiffe_id".
+func isJoinTokenEntry(entry spireapi.Entry) bool {
+	if !strings.Contains(entry.ParentID.String(), joinTokenSpiffeSubstr) {
+		return false
+	}
+	for _, selector := range entry.Selectors {
+		if selector.Type == joinTokenSelectorType {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Is this code used?

Same with the definition joinTokenSelectorType

Reply:
good catch, forgot to use the helper. Fixed now.